atomix/cluster/src/main/java/io/atomix/cluster/protocol/SwimMembershipProtocol.java

```diff
@@ -523,6 +525,10 @@ private void probe(final ImmutableMember member) {
                   requestProbes(swimMember.copy());
                 }
               }
+              swimScheduler.schedule(
```

Comment:
Should we wait for the response to schedule the next probe? Worst case, if the remote member is dead, we wait probeTimeout + probeInterval  time to send the next probe. This will increase the latency to detect failures of other nodes. Right?

Reply:
I think we discussed that, that the current impl was not ideal. Actually I would expect that the timeout is much lower. We could also say on timeout we directly trigger the next probe.

Reply:
The current impl is not ideal because it is scheduled at fixed rate. Doesn't sending the probe and immediately scheduling the next one already reduce that problem? I don't know clearly what will be the impact of both approaches (Any comments @saig0 ). If we wait for the timeout, I am afraid we increase the failure detection latency. Decreasing the timeout would probably help, but it might unnecessarily detect slow nodes as failed. That said, 2s timeout is too much. This timeout should be based on expected RTT in the network. 
I'm ok with accepting this implementation and revisit this if we find any problem in future.

Reply:
![swim](https://user-images.githubusercontent.com/2758593/85676118-981c6900-b6c6-11ea-9c04-b231113ebcbf.png)
Extract from http://www.cs.cornell.edu/info/projects/spinglass/public_pdfs/swim.pdf page 4. So yes it should be smaller then the ping interval.

Example how it is done on other implementations 

* Hashicorp uses 500 ms timeout https://github.com/hashicorp/memberlist/blob/master/config.go#L288 
* scalecube uses 100ms https://github.com/scalecube/scalecube-cluster

Reply:
@deepthidevaki Should I create a follow up for the timeout lowering?

Reply:
Follow up https://github.com/zeebe-io/zeebe/issues/4827

Reply:
:+1: